src/lib/eventListeners.js

```diff
@@ -31,24 +32,46 @@ export default {
       componentsMap.set(component, components)
     }
 
-    components.add(cb)
+    components.add({ cb, priority })
+    cache.delete(event) // Invalidate the cache when a new callback is added
   },
+
   executeListeners(event, params) {
     const componentsMap = eventsMap.get(event)
-    if (componentsMap) {
-      componentsMap.forEach((component) => {
-        component.forEach((cb) => {
-          cb(params)
-        })
+    if (!componentsMap) {
+      return true // No listeners so execution can be considered successful
+    }
+
+    if (!cache.has(event)) {
+      // collect all callbacks and sort them by priority
+      const allCallbacks = []
+      componentsMap.forEach((components, component) => {
+        components.forEach((callbackObj) => allCallbacks.push({ ...callbackObj, component }))
       })
```

Comment:
this may have some performance impact - forEach inside a forEach
we should change this to a for or while loop probably and make sure that the nested loop is really necessary

Reply:
replaced with `for ... of`. I think it is faster than `forEach`. I could not get rid of the nested loop without adding intermediate arrays.